src/nvim/option.c

```diff
@@ -3738,6 +3738,10 @@ did_set_string_option (
       if (varp == &p_enc && *curbuf->b_p_keymap != NUL)
         (void)keymap_init();
 
+      if (varp == &p_enc) {
+        ui_update_encoding();
```

Comment:
Drat, brainfart – @justinmk is right, I confused `encoding` and `fileencoding`. Too much Ruby lately :). Disregard my above comment, please!